### PR TITLE
Add stages in Jenkin to build blockchain connector and run proxy model tests

### PR DESCRIPTION
**Pull request by Ram-srini**
Signed-off-by: Ramakrishna Srinivasamurthy <ramakrishna.srinivasamurthy@intel.com>

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -67,15 +67,30 @@ pipeline {
...
+                sh 'docker-compose -f ci/docker-compose-proxy-model-build.yaml build'
+            }
+        }
+
+        stage('Run Avalon proxy Model Tests') {
+            steps {
+                sh '''
+                   ./scripts/start_fabric.sh -u
+                   INSTALL_TYPE="" ./bin/run_tests -p
+                   '''
```
Comment (pankajgoyal2) [inline Jenkinsfile:93]: Do we need to terminate all the containers for peers, orderers at the end of job?

File: Jenkinsfile [hunk 2]
```diff
@@ -67,15 +67,30 @@ pipeline {
...
-                sh 'INSTALL_TYPE="" ./bin/run_tests'
+                sh 'docker-compose -f ci/docker-compose-proxy-model-build.yaml build'
+            }
+        }
+
+        stage('Run Avalon proxy Model Tests') {
+            steps {
+                sh '''
+                   ./scripts/start_fabric.sh -u
+                   INSTALL_TYPE="" ./bin/run_tests -p
```
Comment (pankajgoyal2) [inline Jenkinsfile:92]: shouldn't it be "./bin/run_tests -f"?

File: ci/avalon-direct-model-tests.yaml [hunk 3]
```diff
@@ -15,15 +15,12 @@
...
-    build:
-      context: .
-    environment:
-      - http_proxy
-      - https_proxy
-      - no_proxy
     working_dir: "/project/avalon/"
+    volumes:
+      - ~/mywork/vars/keyfiles:/keyfiles
+      - ~/mywork/vars/profiles/mychannel_network_for_pysdk.json:/project/avalon/sdk/avalon_sdk/fabric/network.json
```
Comment (pankajgoyal2) [inline ci/avalon-direct-model-tests.yaml:23]: Why to copy mywork directory contents to this yaml file which is for direct model tests?

File: Jenkinsfile (hunk 2 above)
Comment (Ram-srini) [reply Jenkinsfile:92]: Changed to -p in run_tests

Comment (pankajgoyal2) [reply Jenkinsfile:92]: ok

File: Jenkinsfile [hunk 4]
```diff
@@ -92,6 +107,10 @@ pipeline {
     }
 
     post {
+        always {
+            echo 'Cleaning up fabric network'
+            sh './scripts/start_fabric.sh -c'
```
Comment (danintel) [inline Jenkinsfile:112]: I would bring down Fabric with `./scripts/start_fabric.sh -d` before running `-c` to clean it up. 
It seems cleaner to do that.
Is there a reason not to run `./scripts/start_fabric.sh -d`  ?

File: bin/run_tests [hunk 5]
```diff
@@ -24,8 +24,37 @@ PROJECT_DIR=$(cd $(dirname $(dirname $0)) && pwd)
 # Make sawtooth scripts accessible
```
Comment (danintel) [inline bin/run_tests:24]: This comment seems obsolete.  I think it should say "Make Avalon" not "Make sawtooth"

File: bin/run_tests [hunk 6]
```diff
@@ -24,8 +24,37 @@ PROJECT_DIR=$(cd $(dirname $(dirname $0)) && pwd)
...
+while getopts "dph" opt; do
+    case $opt in
+        d)
+            DIRECT_MODEL_TESTS=1
+            ;;
+        p)
+            PROXY_MODEL_TESTS=1
+            ;;
+        h)
+            echo "Usage:"
```
Comment (danintel) [inline bin/run_tests:49]: Before the `Usage:` line add a line explaining what `run_tests` does.  Something like this (reword or correct if needed):

`echo "Run Avalon CI tests in Docker"`

File: Jenkinsfile (hunk 4 above)
Comment (Ram-srini) [reply Jenkinsfile:112]: -c will bring down the fabric container and also cleanup of work_dir where fabric certificates stored.

File: tools/run_proxy_model_tests.sh [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+yell() {
+    echo "$0: $*" >&2;
+}
+
+die() {
+    yell "$*"
+    exit 111
```
Comment (manojgop) [inline tools/run_proxy_model_tests.sh:23]: What does exit 111 mean ?

File: tools/run_proxy_model_tests.sh [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+try() {
+    "$@" || die "test failed: $*"
+}
+
+
+SCRIPTDIR="$(dirname $(readlink --canonicalize ${BASH_SOURCE}))"
+SRCDIR="$(realpath ${SCRIPTDIR}/..)"
+generic_client_path="${TCF_HOME}/examples/apps/generic_client"
+
+sleep 50s
```
Comment (manojgop) [inline tools/run_proxy_model_tests.sh:35]: Why do we need `sleep 50` ?

File: tools/run_proxy_model_tests.sh [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+yell "Start testing fabric generic client for heart disease eval workload ................"
+yell "#------------------------------------------------------------------------------------------------"
+try $generic_client_path/fabric_generic_client.py --blockchain fabric \
+        --workload_id "heart-disease-eval" \
+        --in_data "Data: 25 10 1 67  102 125 1 95 5 10 1 11 36 1" -o
+
+yell "#------------------------------------------------------------------------------------------------"
+yell "#------------------------------------------------------------------------------------------------"
+
+sleep 10s
```
Comment (manojgop) [inline tools/run_proxy_model_tests.sh:52]: Why do we need `sleep 10` ?

File: bin/run_tests (hunk 5 above)
Comment (Ram-srini) [reply bin/run_tests:24]: changed to avalon.

File: bin/run_tests (hunk 6 above)
Comment (Ram-srini) [reply bin/run_tests:49]: Done

**Pull request merged**